### PR TITLE
test(node affinity): verify volume provisioning by adding non existing node-affinity rule

### PR DESCRIPTION
**Pull request by mittachaitu**
## Pull Request template

**Why is this PR required? What issue does it fix?**:
This PR is required to add negative test cases of
node-affinity feature.

**What this PR does?**:
 This PR adds the following test cases:
- Provision volume by adding non-existing node-affinity rule and
  verify that volume provisioning should fail.
- Add a non-existing node-affinity rule to one of the nodes and
  verify that volume provisioning should succeed

**Does this PR require any upgrade changes?**:
No

**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Any additional information for your reviewer?** : 
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [ ] Fixes #<issue number>
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Has the change log section been updated? 
- [ ] Commit has unit tests
- [x] Commit has integration tests
- [ ] (Optional) Are upgrade changes included in this PR? If not, mention the issue/PR to track: 
- [ ] (Optional) If documentation changes are required, which issue on https://github.com/openebs/openebs-docs is used to track them:

### CODE REVIEW

File: provisioner/provisioner.go [hunk 1]
```diff
@@ -231,7 +235,8 @@ func (p *Provisioner) validateNodeAffinityRules() error {
 
 	if len(nodeList) == 0 {
 		return errors.Errorf(
-			"No matching nodes found for given affinity rules (%s)",
+			"%s (%s)",
+			NodeAffinityRulesMismatchEvent,
```
Comment (mynktl) [inline provisioner/provisioner.go]: nit: this can be merged in one line

File: tests/k8s_utils.go [hunk 2]
```diff
@@ -45,11 +48,15 @@ type KubeClient struct {
...
-var Client *KubeClient
+var (
+	// Client for KubeClient
+	Client *KubeClient
 
-// encoder to print object in yaml format
-var encoder runtime.Encoder
+	// encoder to print object in yaml format
+	encoder          runtime.Encoder
+	defaultChunkSize = int64(500)
```
Comment (mynktl) [inline tests/k8s_utils.go:63]: better to add comment here

File: tests/node_affinity_test.go [hunk 3]
```diff
@@ -207,18 +238,124 @@ var _ = Describe("TEST NODE AFFINITY FEATURE", func() {
...
+			By("creating above pvc")
+			err = Client.createPVC(pvcObj, false)
+			Expect(err).To(BeNil(), "while creating pvc {%s} in namespace {%s}", pvcName, applicationNamespace)
+
+			pvcObj, err = Client.getPVC(pvcObj.Namespace, pvcObj.Name)
+			Expect(err).To(BeNil(), "while fetching PVC %s in namespace %s", pvcObj.Name, pvcObj.Namespace)
+
+			var isExpectedEventExist bool
+			for retries := 0; retries < maxRetryCount; retries++ {
+
```
Comment (mynktl) [inline tests/node_affinity_test.go]: nit: new line not required here

File: provisioner/provisioner.go (hunk 1 above)
Comment (mittachaitu) [reply provisioner/provisioner.go]: Yeah, missed it I will try to not repeat this again

File: tests/k8s_utils.go (hunk 2 above)
Comment (mittachaitu) [reply tests/k8s_utils.go:63]: updated

File: tests/node_affinity_test.go (hunk 3 above)
Comment (mittachaitu) [reply tests/node_affinity_test.go]: remove new line

File: tests/k8s_utils.go [hunk 4]
```diff
@@ -199,20 +215,26 @@ func (k *KubeClient) waitForPVCBound(pvc, ns string) (corev1.PersistentVolumeCla
...
 	_, err := k.CoreV1().PersistentVolumeClaims(pvc.Namespace).Create(pvc)
 	if err != nil {
 		if !k8serrors.IsAlreadyExists(err) {
 			return err
 		}
 	}
-	_, err = k.waitForPVCBound(pvc.Name, pvc.Namespace)
-	return err
+
+	if shouldWaitForProvision {
```
Comment (mynktl) [inline tests/k8s_utils.go]: since we have a different function to wait for PVC bound, let's remove this from `createPVC` function.

Comment (mittachaitu) [reply tests/k8s_utils.go]: Update in all respective places

**Review by mynktl — APPROVED** [review]

**Pull request merged**